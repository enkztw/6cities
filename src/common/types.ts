export type property = `apartment` | `room` | `house` | `hotel`;
export type id = number
export type city = `Paris` | `Cologne` | `Brussels` | `Amsterdam` | `Hamburg` | `Dusseldorf`
export type sort = `rating` | `id` | `toLow` | `toHigh`


