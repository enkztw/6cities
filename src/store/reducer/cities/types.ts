enum Types {
  SET_CITY = `SET_CITY`
}

export default Types;
