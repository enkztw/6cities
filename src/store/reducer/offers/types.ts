enum Types {
  SET_OFFER = `SET_OFFER`,
  SET_COMMENTS = `SET_COMMENTS`,
  SET_OFFERS = `SET_OFFERS`,
  SET_FAVORITES = `SET_FAVORITES`,
  SET_NEARBY = `SET_NEARBY`,
  SET_SORTING = `SET_SORTING`
}

export default Types;
