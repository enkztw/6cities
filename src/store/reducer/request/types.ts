enum Types {
  SET_REQUEST = `SET_REQUEST`,
}

export default Types;

