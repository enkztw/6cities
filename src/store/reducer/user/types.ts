enum Types {
  SET_AUTH = `SET_AUTH`
}

export default Types;

