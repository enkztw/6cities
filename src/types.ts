// import OperationStatus from './const';

export type property = `apartment` | `room` | `house` | `hotel`;
// export type status = OperationStatus.SUCCESS |
export type id = number
export type city = `Paris` | `Cologne` | `Brussels` | `Amsterdam` | `Hamburg` | `Dusseldorf`


