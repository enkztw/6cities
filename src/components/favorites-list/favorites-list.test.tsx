import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';
import configureStore from 'redux-mock-store';

import {Offer} from '../../types';
import FavoritesList from './favorites-list';


const mockStore = configureStore([]);

const favorites: Offer[] = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 1.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 100,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 2,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    previewImage: `img/apartment-02.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 2.8,
    type: `room`,
    bedrooms: 2,
    maxAdults: 4,
    price: 140,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8
    }
  },
  {
    id: 3,
    city: {
      name: `Paris`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    previewImage: `img/apartment-02.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `room`,
    bedrooms: 2,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8
    }
  }
];

it(`<FavoritesList /> should have been rendered`, () => {
  const store = mockStore({
    offers: {
      favorites
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <FavoritesList favorites={favorites}/>
        </Router>
      </Provider>
  );

  expect(tree).toMatchSnapshot();
});
