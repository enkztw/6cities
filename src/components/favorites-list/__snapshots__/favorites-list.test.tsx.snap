// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoritesList /> should be rendered 1`] = `
<ul
  className="favorites__list"
>
  <li
    className="favorites__locations-items"
    key="Amsterdam"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <Connect(Card)
        bedrooms={3}
        city={
          Object {
            "location": Object {
              "latitude": 52.370216,
              "longitude": 4.895168,
              "zoom": 10,
            },
            "name": "Amsterdam",
          }
        }
        description="A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam."
        goods={
          Array [
            "Heating",
            "Kitchen",
            "Cable TV",
            "Washing machine",
            "Coffee machine",
            "Dishwasher",
          ]
        }
        host={
          Object {
            "avatarUrl": "img/avatar-angelina.jpg",
            "id": 3,
            "isPro": true,
            "name": "Angelina",
          }
        }
        id={1}
        images={
          Array [
            "img/apartment-02.jpg",
            "img/apartment-03.jpg",
          ]
        }
        isFavorite={false}
        isPremium={false}
        key="1"
        location={
          Object {
            "latitude": 52.35514938496378,
            "longitude": 4.673877537499948,
            "zoom": 8,
          }
        }
        maxAdults={4}
        mini={true}
        previewImage="img/apartment-01.jpg"
        price={100}
        rating={1.8}
        title="Beautiful & luxurious studio at great location"
        type="apartment"
      />
      <Connect(Card)
        bedrooms={2}
        city={
          Object {
            "location": Object {
              "latitude": 52.370216,
              "longitude": 4.895168,
              "zoom": 10,
            },
            "name": "Amsterdam",
          }
        }
        description="A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam."
        goods={
          Array [
            "Heating",
            "Kitchen",
            "Cable TV",
            "Washing machine",
            "Coffee machine",
            "Dishwasher",
          ]
        }
        host={
          Object {
            "avatarUrl": "img/avatar-angelina.jpg",
            "id": 3,
            "isPro": true,
            "name": "Angelina",
          }
        }
        id={2}
        images={
          Array [
            "img/apartment-01.jpg",
            "img/apartment-03.jpg",
          ]
        }
        isFavorite={false}
        isPremium={false}
        key="2"
        location={
          Object {
            "latitude": 52.3909553943508,
            "longitude": 4.85309666406198,
            "zoom": 8,
          }
        }
        maxAdults={4}
        mini={true}
        previewImage="img/apartment-02.jpg"
        price={140}
        rating={2.8}
        title="Beautiful & luxurious studio at great location"
        type="room"
      />
    </div>
  </li>
  <li
    className="favorites__locations-items"
    key="Paris"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Paris
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <Connect(Card)
        bedrooms={2}
        city={
          Object {
            "location": Object {
              "latitude": 52.370216,
              "longitude": 4.895168,
              "zoom": 10,
            },
            "name": "Paris",
          }
        }
        description="A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam."
        goods={
          Array [
            "Heating",
            "Kitchen",
            "Cable TV",
            "Washing machine",
            "Coffee machine",
            "Dishwasher",
          ]
        }
        host={
          Object {
            "avatarUrl": "img/avatar-angelina.jpg",
            "id": 3,
            "isPro": true,
            "name": "Angelina",
          }
        }
        id={3}
        images={
          Array [
            "img/apartment-01.jpg",
            "img/apartment-03.jpg",
          ]
        }
        isFavorite={false}
        isPremium={false}
        key="3"
        location={
          Object {
            "latitude": 52.3909553943508,
            "longitude": 4.85309666406198,
            "zoom": 8,
          }
        }
        maxAdults={4}
        mini={true}
        previewImage="img/apartment-02.jpg"
        price={120}
        rating={4.8}
        title="Beautiful & luxurious studio at great location"
        type="room"
      />
    </div>
  </li>
</ul>
`;
